app/src/main/java/org/jellyfin/mobile/utils/LifecycleAwareHandler.kt

```diff
@@ -0,0 +1,28 @@
+package org.jellyfin.mobile.utils
+
+import android.os.Handler
+import android.os.Looper
+import android.os.Message
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+
+class LifecycleAwareHandler(lifecycle: Lifecycle, callback: Callback) : Handler(Looper.getMainLooper()), LifecycleObserver {
```

Comment:
Why not something like:
```suggestion
class LifecycleAwareHandler(lifecycle: Lifecycle, callback: (message: Message) -> Unit) : Handler(Looper.getMainLooper()), LifecycleObserver {
```

That way it will work the same as before on the call side

Reply:
Hm yeah, could do that. However, with Kotlin 1.4, it should work without that change.

Reply:
_Upgrade all the things_

Want to change/upgrade in this PR or later?

Reply:
> _Upgrade all the things_
> 
> Want to change/upgrade in this PR or later?

Think I'll keep it like that for now, I plan to upgrade to Kotlin 1.4 once a few days have passed and it's fully stable.